pkg/operator/resourcesynccontroller/core.go

```diff
@@ -55,18 +62,12 @@ func CombineCABundleConfigMaps(destinationConfigMap ResourceLocation, lister cor
 
 	caBytes, err := crypto.EncodeCertificates(finalCertificates...)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
-	cm := &corev1.ConfigMap{
-		ObjectMeta: certrotation.NewTLSArtifactObjectMeta(
-			destinationConfigMap.Name,
-			destinationConfigMap.Namespace,
-			additionalAnnotations,
-		),
-		Data: map[string]string{
-			"ca-bundle.crt": string(caBytes),
-		},
+	modified := additionalAnnotations.EnsureTLSMetadataUpdate(&cm.ObjectMeta)
+	cm.Data = map[string]string{
+		"ca-bundle.crt": string(caBytes),
 	}
-	return cm, nil
+	return cm, modified, nil
```

Comment:
How is the bool return value supposed to be interpreted? It's unclear to me what it means when EnsureTLSMetadataUpdate returns false (and so the returned bool value is false) but Data has been mutated.

Reply:
`modified` here returns if [update is necessary](https://github.com/openshift/cluster-kube-apiserver-operator/pull/1812/files#diff-0af15f182aea9898e784fc589175ef4c2411c9391bb5a1f41327ba7c3e2172f5R367):
```
requiredConfigMap, updateRequired, err := resourcesynccontroller.CombineCABundleConfigMaps(
```

It means either metadata or contents has changed. I think I missed "content has changed, but not the metadata", so I'll add some unittests to verify that